ako-gateway-api/k8s/validator.go

```diff
@@ -166,8 +157,11 @@ func isValidListener(key string, gateway *gatewayv1.Gateway, gatewayStatus *gate
 	for _, gatewayInNamespace := range gatewayNsList {
 		if gateway.Name != gatewayInNamespace.Name {
 			for _, gwListener := range gatewayInNamespace.Spec.Listeners {
-				if *listener.Hostname == *gwListener.Hostname || isRegexMatch(string(*listener.Hostname), string(*gwListener.Hostname), key) || isRegexMatch(string(*gwListener.Hostname), string(*listener.Hostname), key) {
-					utils.AviLog.Errorf("key: %s, msg: Hostname overlaps or is same as an existing gateway %s hostname %s", key, gatewayInNamespace.Name, *gwListener.Hostname)
+				if gwListener.Hostname == nil {
+					continue
+				}
+				if listener.Hostname != nil && *listener.Hostname == *gwListener.Hostname {
+					utils.AviLog.Errorf("key: %s, msg: Hostname is same as an existing gateway %s hostname %s", key, gatewayInNamespace.Name, *gwListener.Hostname)
 					defaultCondition.
 						Message("Hostname overlaps or is same as an existing gateway hostname").
```

Comment:
We can change this message now as it only error out when there is an exact match and no overlapping